### PR TITLE
Docfix

### PR DESCRIPTION
**Pull request by chummels**
Updates viz docs for yt4.

### CODE REVIEW

File: doc/source/visualizing/callbacks.rst [hunk 1]
```diff
@@ -864,9 +870,9 @@ Overplot the Path of a Ray
 
     Adds a line representing the projected path of a ray across the plot.  The
     ray can be either a
-    :class:`~yt.data_objects.selection_data_containers.YTOrthoRay`,
-    :class:`~yt.data_objects.selection_data_containers.YTRay`, or a
-    :class:`~trident.light_ray.LightRay`
+    :class:`~yt.data_objects.selection_objects.ray.YTOrthoRay`,
+    :class:`~yt.data_objects.selection__objects.ray.YTRay`, or a
```
Comment (cphyc) [inline doc/source/visualizing/callbacks.rst:874]:
```suggestion
    :class:`~yt.data_objects.selection_objects.ray.YTRay`, or a
```

**Review by cphyc — APPROVED** [review]
LGTM!

**Review by brittonsmith — APPROVED** [review]

File: doc/source/analyzing/fields.rst [hunk 2]
```diff
@@ -21,19 +21,19 @@ What are fields?
...
-to select, given the field type. Possible field names include *density*,
-*velocity_x* or *pressure* --- these three fields are examples of field names
+The second element of field tuples, the ``field_name``, denotes the specific field
+to select, given the field type. Possible field names include ``density``,
+``velocity_x`` or ``pressure`` --- these three fields are examples of field names
 that might be used for a fluid defined on a mesh. Examples of particle fields
-include *particle_mass*, *particle_position*, or *particle_velocity_x*. In
-general, particle field names are prefixed by "particle\_", which makes it easy
+include ``particle_mass`` ``particle_position`` or ``particle_velocity_x`` In
+general, particle field names are prefixed by ``particle\_``, which makes it easy
```
Comment (neutrinoceros) [inline doc/source/analyzing/fields.rst:36]: I don't think escaping the underscore should be necessary (or desired) now that you've correctly backtick'ed it
```suggestion
general, particle field names are prefixed by ``particle_``, which makes it easy
```

File: doc/source/visualizing/colormaps/index.rst [hunk 3]
```diff
@@ -144,12 +144,12 @@ callback:
 .. code-block:: python
 
     ds = yt.load("IsolatedGalaxy/galaxy0030/galaxy0030")
-    p = yt.ProjectionPlot(ds, "z", "density")
+    p = yt.ProjectionPlot(ds, "z", ("gas", "density"))
 
-    p.set_cmap(field="density", cmap="jet")
+    p.set_cmap(field=("gas", "density"), cmap="jet")
```
Comment (neutrinoceros) [inline doc/source/visualizing/colormaps/index.rst:149]: I suggest taking the opportunity to obliterate jet from our docs :)
```suggestion
    p.set_cmap(field=("gas", "density"), cmap="viridis")
```

File: doc/source/visualizing/manual_plotting.rst [hunk 4]
```diff
@@ -44,14 +44,14 @@ of any data two-dimensional data object:
...
-   proj = ds.proj("density", 0)
+   _, c = ds.find_max(("gas", "density"))
+   proj = ds.proj(("gas", "density"), 0)
 
    width = (10, "kpc")  # we want a 1.5 mpc view
    res = [1000, 1000]  # create an image with 1000x1000 pixels
    frb = proj.to_frb(width, res, center=c)
 
-   plt.imshow(np.array(frb["density"]))
+   plt.imshow(np.array(frb[("gas", "density")]))
```
Comment (neutrinoceros) [inline doc/source/visualizing/manual_plotting.rst:54]:
```suggestion
   plt.imshow(np.array(frb["gas", "density"]))
```

File: doc/source/visualizing/manual_plotting.rst [hunk 5]
```diff
@@ -93,7 +93,7 @@ using them matters.
    frb = slc.to_frb((20, "kpc"), 512)
    frb.apply_gauss_beam(nbeam=30, sigma=2.0)
    frb.apply_white_noise(5e-23)
-   plt.imshow(frb["density"].d)
+   plt.imshow(frb[("gas", "density")].d)
```
Comment (neutrinoceros) [inline doc/source/visualizing/manual_plotting.rst:96]:
```suggestion
   plt.imshow(frb["gas", "density"].d)
```

File: doc/source/visualizing/manual_plotting.rst [hunk 6]
```diff
@@ -143,21 +143,21 @@ created from a index by calling ``pf.ortho_ray(axis, center)``.
...
+   _, c = ds.find_max(("gas", "density"))
    ax = 0  # take a line cut along the x axis
 
    # cutting through the y0,z0 such that we hit the max density
    ray = ds.ortho_ray(ax, (c[1], c[2]))
 
    # Sort the ray values by 'x' so there are no discontinuities
    # in the line plot
-   srt = np.argsort(ray["x"])
+   srt = np.argsort(ray[("index", "x")])
```
Comment (neutrinoceros) [inline doc/source/visualizing/manual_plotting.rst:154]:
```suggestion
   srt = np.argsort(ray["index", "x"])
```

File: doc/source/visualizing/manual_plotting.rst [hunk 7]
```diff
@@ -143,21 +143,21 @@ created from a index by calling ``pf.ortho_ray(axis, center)``.
...
 
    plt.subplot(211)
-   plt.semilogy(np.array(ray["x"][srt]), np.array(ray["density"][srt]))
+   plt.semilogy(np.array(ray[("index", "x")][srt]), np.array(ray[("gas", "density")][srt]))
```
Comment (neutrinoceros) [inline doc/source/visualizing/manual_plotting.rst:157]:
```suggestion
   plt.semilogy(np.array(ray["index", "x"][srt]), np.array(ray["gas", "density"][srt]))
```

File: doc/source/visualizing/manual_plotting.rst [hunk 8]
```diff
@@ -143,21 +143,21 @@ created from a index by calling ``pf.ortho_ray(axis, center)``.
...
    plt.ylabel("density")
    plt.subplot(212)
-   plt.semilogy(np.array(ray["x"][srt]), np.array(ray["temperature"][srt]))
+   plt.semilogy(np.array(ray[("index", "x")][srt]), np.array(ray[("gas", "temperature")][srt]))
```
Comment (neutrinoceros) [inline doc/source/visualizing/manual_plotting.rst:160]:
```suggestion
   plt.semilogy(np.array(ray["index", "x"][srt]), np.array(ray["gas", "temperature"][srt]))
```

File: doc/source/visualizing/sketchfab.rst [hunk 9]
```diff
@@ -57,14 +57,14 @@ value.  For example:
...
-   surface = ds.surface(sphere, "density", 1e-27)
+   surface = ds.surface(sphere, ("gas", "density"), 1e-27)
 
 This object, ``surface``, can be queried for values on the surface.  For
 instance:
 
 .. code-block:: python
 
-   print(surface["temperature"].min(), surface["temperature"].max())
+   print(surface[("gas", "temperature")].min(), surface[("gas", "temperature")].max())
```
Comment (neutrinoceros) [inline doc/source/visualizing/sketchfab.rst:67]:
```suggestion
   print(surface["gas", "temperature"].min(), surface["gas", "temperature"].max())
```

File: doc/source/visualizing/sketchfab.rst [hunk 10]
```diff
@@ -246,18 +249,22 @@ to output one more type of variable on your surfaces.  For example:
 
 
     def emissivity(field, data):
-        return data["density"] * data["density"] * np.sqrt(data["temperature"])
+        return (
+            data[("gas", "density")]
+            * data[("gas", "density")]
+            * np.sqrt(data[("gas", "temperature")])
```
Comment (neutrinoceros) [inline doc/source/visualizing/sketchfab.rst:255]:
```suggestion
            data["gas", "density"] ** 2
            * np.sqrt(data["gas", "temperature"])
```

File: doc/source/visualizing/streamlines.rst [hunk 11]
```diff
@@ -129,7 +129,7 @@ Example Script
     streamlines = Streamlines(ds, ds.domain_center)
     streamlines.integrate_through_volume()
     stream = streamlines.path(0)
-    matplotlib.pylab.semilogy(stream["t"], stream["density"], "-x")
+    matplotlib.pylab.semilogy(stream["t"], stream[("gas", "density")], "-x")
```
Comment (neutrinoceros) [inline doc/source/visualizing/streamlines.rst:132]:
```suggestion
    matplotlib.pylab.semilogy(stream["t"], stream["gas", "density"], "-x")
```

**Review by neutrinoceros — CHANGES_REQUESTED** [review]
This is great, thanks. I sprinkled a couple suggestions but they're all the same: remove superfluous parentheses in field tuple keys. The code you wrote is _valid_ and _works_ the same, but I really think we should keep it as simple as possible, given that tuple keys represent a increase in user code complexity already.

**Review by neutrinoceros — APPROVED** [review]
pre-commit.ci autofix

Edit: got it wrong and apparently the bot doesn't parse updates

**Comment by neutrinoceros** [comment]
pre-commit.ci autofix

**Pull request merged**